internal/k8s/annotations.go

```diff
@@ -215,24 +219,69 @@ func MinProtoVersion(version string) envoy_api_v2_auth.TlsParameters_TlsProtocol
 	}
```

Comment:
`MinProtoVersion` isn't related to Kubernetes or annotations. I think it makes more sense to put this helper into the `envoy` package, but that causes an import cycle because `envoy` depends on `dag`. I suppose it could go into `dag` in a pinch.